### PR TITLE
Fix types for epicStrategyBuilder

### PR DESCRIPTION
**Pull request by sergej-tschak**
- PR with @benj56 improvements for #7

### CODE REVIEW

**Review by AlGvrl — APPROVED** [review]

File: src/entities/absenceTypes/absenceTypesEpic.ts [hunk 1]
```diff
@@ -16,14 +15,14 @@ const requestInitEpic = (action$: Observable<Action>) =>
...
-const requestSyncEpic = syncStrategy<AbsenceType>(
+const requestSyncEpic = syncStrategy(
   absenceTypesSlice.actions.requestSync,
   timeTacApi.absenceTypes.readRaw.bind(timeTacApi.absenceTypes), // Issue see SP-559
   absenceTypesSlice.actions.upsert,
   absenceTypesSlice.actions.remove,
   absenceTypesSlice.actions.requestSync,
   initialLoadSlice.actions.singleResouceCompleted,
-  { absenceTypes: true } as BlockingResources,
+  { absenceTypes: true },
```
Comment (haubitzer) [inline src/entities/absenceTypes/absenceTypesEpic.ts:25]: Please, remove as BlockingResources also from the calls, users and projects ...

File: src/deltaSync/deltaSyncEpic.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ function getActionsForResponse<R>(
   upsertAction: ActionCreatorWithPayload<R[], string>,
   removeAction: ActionCreatorWithPayload<DeletedEntry[], string>
 ) {
-  const hasResults = (response && response.results && response.results.length && upsertAction(response.results as R[])) || false;
+  const hasResults = Boolean(response && response.results && response.results.length && upsertAction(response.results as R[]));
```
Comment (haubitzer) [inline src/deltaSync/deltaSyncEpic.ts:82]: This will not work, because now hasResults is always a boolean, and before it was the action or false

**Review by haubitzer — CHANGES_REQUESTED** [review]

File: src/entities/absenceTypes/absenceTypesEpic.ts (hunk 1 above)
Comment (sergej-tschak) [reply src/entities/absenceTypes/absenceTypesEpic.ts:25]: removed

File: src/deltaSync/deltaSyncEpic.ts (hunk 2 above)
Comment (sergej-tschak) [reply src/deltaSync/deltaSyncEpic.ts]: This one was rolled back

**Review by haubitzer — APPROVED** [review]

**Review by benj56 — APPROVED** [review]

**Pull request merged**